contrib/repairStatus.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import print_function
+import json
+import argparse
+import requests
+import re
+import logging as log
+from datetime import datetime, timedelta
+
+
+def generate_range_repair_query (node, start_time, current_time):
+  query_json = {
+    "query": {
+        "filtered": {
+            "query": {
+                "bool": {
+                    "should": [
+                        {
+                            "query_string": {
+                                "query": 'level:"info" AND (*)'
+                            }
+                        }
+                    ]
+                }
+            },
+            "filter": {
+                "bool": {
+                    "must": [
+                        {
+                            "range": {
+                                "@timestamp": {
+                                    "from": start_time,
+                                    "to": current_time
+                                }
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'host:("' + node + '")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'method:("syncComplete")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        },
+                        {
+                            "fquery": {
+                                "query": {
+                                    "query_string": {
+                                        "query": 'message:("is fully synced")'
+                                    }
+                                },
+                                "_cache": True
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+    }
+  }
+
+  return json.dumps(query_json)
+
+
+# Execute an ES _count to return the number of ranges that have finished repairing
+def calculate_range_repairs(host, port, node, start_time, current_time):
+  url = 'http://%(host)s:%(port)s/_count' % dict(host=host, port=port)
+  query = generate_range_repair_query(node, start_time, current_time)
+  
+  log.info("Ranges completed: curl %(url)s -d '%(query)s'" % dict(url=url, query=query)) 
+  r = requests.get(url, data=query)
+  
+  if r.status_code == 200:
+    response = json.loads(r.text)
+    return response['count']
```

Comment:
Checking the return code this way means that when the remote end encounters any sort of error, you won't see what it was.  I think it would be better to just blow up, like `raise ValueError("invalid status code returned: {code} {text}".format(code=response.code, text=response.text)).  This way you have something to look at if the query goes wrong because the cluster is in bad shape, and more diagnostics to act on later.